forest/intake_loader.py

```diff
@@ -0,0 +1,395 @@
+'''
+Module for loading data from the Pangeo CMIP6n intake catalogue (usng the intake-esm
+loader) created by NCAR.
+'''
+
+from datetime import datetime
+from collections import namedtuple
+import functools
+
+import bokeh
+import intake
+import iris
+import numpy
+
+from forest import geo, gridded_forecast
+
+# Location of the Pangeo-CMIP6 intake catalogue file.
+URL = 'https://raw.githubusercontent.com/NCAR/intake-esm-datastore/master/catalogs/pangeo-cmip6.json'
+
+@functools.lru_cache(maxsize=64)
+def _get_intake_vars(
+        experiment_id,
+        table_id,
+        grid_label,
+        institution_id,
+        member_id):
+    """
+    Helper function to get a list of variables for this particular combination
+    of parameters. Function is cahced to reduce remote queries.
+    """
+    collection = intake.open_esm_datastore(URL)
+    cat = collection.search(
+        experiment_id=experiment_id,
+        table_id=table_id,
+        grid_label=grid_label,
+        institution_id=institution_id,
+        member_id=member_id,
+    )
+    var_list = cat.unique('variable_id')['variable_id']['values']
+    return var_list
+
+
+@functools.lru_cache(maxsize=16)
+def _load_from_intake(
+        experiment_id,
+        table_id,
+        grid_label,
+        variable_id,
+        institution_id,
+        activity_id,
+        member_id):
+    """
+    Load data from the pangeo CMIP6 intake catalogue.The arguments relate to
+    the CMIP6 parameters of a dataset. The CMIP6 reference is the ESGF servers
+    which can be accessed here:
+    https://esgf-index1.ceda.ac.uk/search/cmip6-ceda/
+    Function is cahced to reduce remote queries.
+    """
+    collection = intake.open_esm_datastore(URL)
+    print('opening catalogue')
+    cat = collection.search(
+        experiment_id=experiment_id,
+        table_id=table_id,
+        grid_label=grid_label,
+        institution_id=institution_id,
+        member_id=member_id,
+        variable_id=variable_id)
+    print('downloading data')
+    dset_dict = cat.to_dataset_dict(
+        zarr_kwargs={'consolidated': True, 'decode_times': False},
+        cdf_kwargs={'chunks': {}, 'decode_times': False})
+    try:
+        parent_source_id = [l1 for l1 in dset_dict.values()][0].parent_source_id
+    except IndexError:
+        import pdb
+        pdb.set_trace()
+    ds_label = f'{activity_id}.{institution_id}.{parent_source_id}.{experiment_id}.{table_id}.{grid_label}'
+    xr = dset_dict[ds_label]
+    print(xr[variable_id])
+    cube = xr[variable_id].to_iris()
+
+    coord_names = [c1.name() for c1 in cube.coords()]
+    if 'air_pressure' in coord_names:
+        cube.coord('air_pressure').convert_units('hPa')
+    return cube[0]  # drop member dimension
+
+
+class IntakeView(object):
```

Comment:
Most of this class is exactly the same as `view.UMView` ... it'd be good if we could avoid that repetition. A few options come to mind (in vague order of preference):

1. Use UMView for both cases (and give it a more generic name), but have it ask the loader for if there are any extra fields that describe the current data.
2. Use UMView for both cases (and give it a more generic name), but when it's constructed have a parameter describing any extra fields.
3. Use inheritence. 😑

Reply:
IntakeView was only specifying custom tool tips, so I've added a method to UMView to specify tooltips, switched to using UMView for intake and specifying the tooltips, and deleted IntakeView. (70c61cb)